app/controllers/users/registrations_controller.rb

```diff
@@ -0,0 +1,26 @@
+class Users::RegistrationsController < Devise::RegistrationsController
+    respond_to :json
+
+    skip_forgery_protection only: [:create]
+  
+    private
+  
+    def respond_with(resource, _opts = {})
+      register_success && return if resource.persisted?
+  
+      register_failed resource
+    end
+  
+    def register_success
+      cookies["CSRF-TOKEN"] = form_authenticity_token
```

Comment:
This seems to be the only thing that was missing from this assignment that should have been updated to be the same as the app/controllers/users/sessions_controller.rb file, line 15. That's a simple fix so not something to stop this whole assignment! Just be sure to look through those instructions carefully to make sure there aren't changes that are requested in multiple files at once!